### PR TITLE
Added Stocks You Follow and Error Component

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added stocks that the user follows to home page with routes to their respective stock pages.
- Added /follows endpoint that retrieves all the stocks that the current user follows.
- Added an Error component and context for convenience.

https://github.com/javidangarcia/stock-zone/assets/89328228/17ee1140-71b4-4e33-8d62-f23378aace34

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Instructions for Starting Express Backend
```
Comment (vandanasub) [inline backend/README.md:1]: nice! always great to have a good Readme

File: frontend/src/components/StocksYouFollow/StocksYouFollow.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        };
+        fetchStocksYouFollow();
+    }, []);
+
+    return (
+        <div className="stocks-you-follow">
+            <h1>Stocks You Follow</h1>
+            <div className="stocks">
+                {
+                    stocks && stocks.map((stock) => {
```
Comment (vandanasub) [inline frontend/src/components/StocksYouFollow/StocksYouFollow.jsx:29]: what if you just do `stocks?.map((stock) => {` instead?

**Review by vandanasub — COMMENTED** [review]
Video in PR description looks great! This is coming together very nicely

File: backend/models/follow.js [hunk 3]
```diff
@@ -12,4 +12,6 @@ export const Follow = sequelize.define("Follow", {
 });
 
 User.belongsToMany(Stock, { through: Follow });
-Stock.belongsToMany(User, { through: Follow });
\ No newline at end of file
+Stock.belongsToMany(User, { through: Follow });
+Follow.belongsTo(Stock);
+Stock.hasMany(Follow);
```
Comment (BriptimusPrimus) [inline backend/models/follow.js:17]: Don't we need `Follow.belongsTo(User);` as well?

File: backend/routes/stocks.js [hunk 4]
```diff
@@ -21,18 +21,22 @@ router.get("/stocks/:ticker", async (req, res) => {
         where: { ticker }
     });
 
-    const user = req.session.user;
+    if (stock) {
```
Comment (BriptimusPrimus) [inline backend/routes/stocks.js:24]: stock is not a boolean. It is better to explicitly check for it not being undefined

File: frontend/src/components/Follow/Follow.jsx [hunk 5]
```diff
@@ -1,50 +1,39 @@
 import "./Follow.css";
 import axios from "axios";
+import { UserContext } from "../App/App";
+import { useContext } from "react";
 
 export default function Follow({ ticker, stockData, setStockData }) {
+    const { setError } = useContext(UserContext);
+
     async function handleFollow() {
+        const url = stockData.following
```
Comment (BriptimusPrimus) [inline frontend/src/components/Follow/Follow.jsx:10]: 👍

File: frontend/src/components/Error/Error.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import "./Error.css";
+import { useState, useEffect, useContext } from "react";
+import { UserContext } from "../App/App";
+
+export default function Error({ error }) {
+    const [errorMessage, setErrorMessage] = useState(error);
```
Comment (BriptimusPrimus) [inline frontend/src/components/Error/Error.jsx:6]: I don't think we need a local state, accessing the global state error should suffice. 
Keeping track of two pieces of state for the same thing makes things more complex.

File: frontend/src/components/App/App.jsx [hunk 7]
```diff
@@ -6,9 +6,13 @@ import StockData from "../StockData/StockData";
 import { BrowserRouter, Routes, Route } from "react-router-dom";
 import SignUpForm from "../SignUpForm/SignUpForm";
 import LoginForm from "../LoginForm/LoginForm";
-import { useState, useEffect } from "react";
+import { useState, useEffect, createContext } from "react";
+import Error from "../Error/Error";
+
+export const UserContext = createContext();
```
Comment (BriptimusPrimus) [inline frontend/src/components/App/App.jsx:12]: React Context. Fancy :)

File: frontend/src/components/SignUpForm/SignUpForm.jsx [hunk 8]
```diff
@@ -47,11 +46,7 @@ export default function SignUpForm({ setUser }) {
                 }, 1000);
             }
 
-            if (response.status === 409) {
-                setError(response.data.error);
-            }
-
-            if (response.status === 400) {
+            if (response.status === 409 || response.status === 400) {
```
Comment (BriptimusPrimus) [inline frontend/src/components/SignUpForm/SignUpForm.jsx:49]: 👍

File: frontend/src/components/StockData/StockData.jsx [hunk 9]
```diff
@@ -87,7 +83,6 @@ export default function StockData(props) {
         <div className="stock-data">
             {isLoading ? (
                 <div className="loading">
-                    <div className="ball"></div>
```
Comment (BriptimusPrimus) [inline frontend/src/components/StockData/StockData.jsx:90]: No spinner?

File: frontend/src/components/StocksYouFollow/StocksYouFollow.jsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+export default function StocksYouFollow() {
+    const [stocks, setStocks] = useState([]);
+    const { setError } = useContext(UserContext);
+
+    useEffect(() => {
+        const fetchStocksYouFollow = async () => {
+            try {
+                const response = await axios.get("http://localhost:3000/follows", { withCredentials: true });
+                setStocks(response.data.stocks);
```
Comment (BriptimusPrimus) [inline frontend/src/components/StocksYouFollow/StocksYouFollow.jsx:15]: What if data is undefined?

File: frontend/src/components/StockData/StockData.jsx (hunk 9 above)
Comment (BriptimusPrimus) [reply frontend/src/components/StockData/StockData.jsx:90]: That's OK. It's not too important to the project. You can add it as a stretch stretch goal at the bottom of the backlog

**Pull request merged**